src/NHibernate/Util/MakeQueryableHelper.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using System.Reflection;
+
+namespace NHibernate.Util
+{
+	internal static class MakeQueryableHelper
+    {
+		private static readonly Lazy<MethodInfo> _implMethod = new Lazy<MethodInfo>(() =>
+			typeof(MakeQueryableHelper).GetMethod(nameof(MakeQueryableImpl), BindingFlags.Static | BindingFlags.NonPublic));
```

Comment:
NHibernate has reflection helpers. Such reflection is usually done as:

```C#
private static readonly MethodInfo _implMethodDefinition =
	ReflectHelper.GetMethodDefinition(() => MakeQueryableHelper.MakeQueryableImpl<object>(null));
```

Better follow the NHibernate usual pattern.